src/MainnetController.sol

```diff
@@ -700,53 +573,14 @@ contract MainnetController is AccessControl {
     {
         _checkRole(RELAYER);
 
-        uint256 maxSlippage = maxSlippages[pool];
-        require(maxSlippage != 0, "MainnetController/max-slippage-not-set");
-
-        ICurvePoolLike curvePool = ICurvePoolLike(pool);
-
-        require(
-            minWithdrawAmounts.length == curvePool.N_COINS(),
-            "MainnetController/invalid-min-withdraw-amounts"
-        );
-
-        // Normalized to provide 36 decimal precision when multiplied by asset amount
-        uint256[] memory rates = curvePool.stored_rates();
-
-        // Aggregate the minimum values of the withdrawn assets (e.g. USD)
-        uint256 valueMinWithdrawn;
-        for (uint256 i = 0; i < minWithdrawAmounts.length; i++) {
-            valueMinWithdrawn += minWithdrawAmounts[i] * rates[i];
-        }
-        valueMinWithdrawn /= 1e18;
-
-        // Check that the aggregated minimums are greater than the max slippage amount
-        require(
-            valueMinWithdrawn >= lpBurnAmount * curvePool.get_virtual_price() * maxSlippage / 1e36,
-            "MainnetController/min-amount-not-met"
-        );
-
-        withdrawnTokens = abi.decode(
-            proxy.doCall(
-                pool,
-                abi.encodeCall(
-                    curvePool.remove_liquidity,
-                    (lpBurnAmount, minWithdrawAmounts, address(proxy))
-                )
-            ),
-            (uint256[])
-        );
-
-        // Aggregate value withdrawn to reduce the rate limit
-        uint256 valueWithdrawn;
-        for (uint256 i = 0; i < withdrawnTokens.length; i++) {
-            valueWithdrawn += withdrawnTokens[i] * rates[i];
-        }
-        valueWithdrawn /= 1e18;
-
-        rateLimits.triggerRateLimitDecrease(
-            RateLimitHelpers.makeAssetKey(LIMIT_CURVE_WITHDRAW, pool),
-            valueWithdrawn
+        withdrawnTokens = CurveLib.removeLiquidity(
```

Comment:
Can we use a struct here too actually? I know we don't have stack too deep but it would be good to have the consistency and naming.

Reply:
fixed